tf2_ros/include/tf2_ros/buffer_interface.h

```diff
@@ -55,11 +55,10 @@ namespace tf2_ros
     return time_msg;
   }
 
-  inline const tf2::TimePoint& fromMsg(const builtin_interfaces::msg::Time & time_msg)
```

Comment:
All right, so here is my understanding of the situation (based on [1], and what I think I understand)

A signature like: inline const tf2::TimePoint& fromMsg(...) is returning a reference to an object.  This is only OK if the object is going to stick around after return from the function, i.e. this is a reference to a member value, or something like that.  If the object is *not* going to stick around, then this can lead to a crash, or (worse) later stack corruption.  We get away with it in https://github.com/ros2/geometry2/pull/7/files#diff-bb6bd2d938a7c0ed4397920b3d645fd3R70 because it is returning a reference to a passed-in object, which has a different lifetime.  However, that does *not* hold true for fromMsg(TimePoint) here, because we are actually creating a new object.  Given that, I think we *must* change this to return an object, and not a reference.  @dhood, that also implies we'll have to fix that line of code, and API, that you referenced.  Can someone else comment about whether I'm making sense here?

[1] http://stackoverflow.com/questions/752658/is-the-practice-of-returning-a-c-reference-variable-evil